salt/states/pkg.py

```diff
@@ -508,8 +508,11 @@ def _find_install_targets(name=None,
         # add it to the kwargs.
         kwargs['refresh'] = refresh
 
+
+    resolve_capabilities = kwargs.get('resolve_capabilities', False) and 'pkg.list_provides' in __salt__
```

Comment:
Why would you check if `pkg.list_provides` is in `__salt__`? It will appear there, once this module gets introspected. Or you expect the method won't appear?

Reply:
I need to check if the package module implements this function.
Currently only zypper.py implements this. yum, apt, etc. do not have this.

(We are here in the package state and not in the zypper module)

Reply:
Why you are concerned about yum/apt/etc from within zypper module? This check makes sense if you are calling `pkg.list_provides` from somewhere else, like Git or Docker module. But inside the Zypper code — how you expect the method that is right above you is not callable?

Reply:
Oh, I hate myself now... :laughing: Sorry, didn't noticed URI changes. Well, this is what you've get if there is no interfaces concept. :unamused: OK, I see. Let's leave it for now as it is. I have an idea how to solve it better in the future, but this would require another solid PR.